feat(README): Update readme with information on our domain (#148)
DHBW-Mailflow/mailserver: README.md

- `ListenerService`: A interface to listen to ports and handle incoming connections.
- `LocalListenerService`: Represents a service to listen to a local port and handle incoming connections.
</details>

<details>
  <summary>job</summary>

  - `Job`: A interface to define a job.
  - `JobExecutionException`: A custom exception to handle job execution errors.
  - `WorkerQueue`: Represents a queue to handle jobs.
</details>

<details>
  <summary>usersettings</summary>

- changepassword
  - `ChangePasswordService`: Represents a service to change the password of a user.
  - `ChangePasswordUseCase`: A interface to handle the changing of the password of a user.
- changesignature
  - `ChangeSignatureService`: Represents a service to change the signature of a user.
  - `ChangeSignatureUseCase`: A interface to handle the changing of the signature of a user.
  - `SettingsBuilder`: A builder to build settings.
- export
  - `ExportUseCase`: A interface to handle the exporting of emails.
  - `MailboxExportService`: Represents a service to export a mailbox.
- `UCCollectionSettings`: A collection of all use cases for the user settings domain.

> ### External Layer

Reviewer: ```suggestion
```

- searchemails
  - `SearchEmailsCLIPrompt`: Represents a command line interface to search emails.
  - `SearchEmailTypesCLIPrompt`: Represents a command line interface to select the type of search.
- showemails
  - `AnswerEmailCLIPrompt`: Represents a command line interface to answer an email.
  - `ReadEmailCLIPrompt`: Represents a command line interface to read an email.
  - `ReadEmailContentCLIPrompt`: Represents a command line interface to read the content of an email.
  - `ShowEmailsCLIPrompt`: Represents a command line interface to show emails.
  - `ShowEmailsTypesCLIPrompt`: Represents a command line interface to select the type of emails to show.
- `ComposeEmailCLIPrompt`: Represents a command line interface to compose an email.
- `LoginCLIPrompt`: Represents a command line interface to log in a user.
- `LogoutCLIPrompt`: Represents a command line interface to log out a user.
- `MarkEmailCLIPrompt`: Represents a command line interface to mark an email.
- `OrganizeEmailsCLIPrompt`: Represents a command line interface to organize emails.
- `RegisterCLIPrompt`: Represents a command line interface to register a user.

</details>

> - `BaseCLIPrompt`: Represents a base command line interface that implements the Server interface.

Reviewer: ```suggestion
- `BaseCLIPrompt`: Represents a base command line interface that implements the Server interface and provides reusable functionality for other CLIPrompts
```

> - `MainCLIPrompt`: Represents the main command line interface that extends the BaseCLIPrompt.

Reviewer: ```suggestion
- `MainCLIPrompt`: Represents the main command line interface which is the CLI-Entrypoint to this application.
```
